internal/clock/clock.go

```diff
@@ -0,0 +1,131 @@
+// Package clock provides an interface for clocks.
+//
+// This is useful for testing code which checks does things based on
+// time; it allows advancing the clock as needed by the tests, rather
+// than needing to race with real time.
+package clock
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"capnproto.org/go/capnp/v3/internal/syncutil"
+)
+
+// A Clock can measure time.
```

Comment:
You don't say! 😆 😜

Reply:
I think this was a case of "having trouble thinking of what to put -> prototyping mode, fix it later."

Reply:
Of course - just teasing 😉